### PR TITLE
Add event for when Eureka registry is updated

### PR DESCRIPTION
**Pull request by macsux**
Adds support for #761

### CODE REVIEW

**Comment by TimHess** [comment]
Given the location of the change it's not immediately clear to me how somebody would take advantage of this change, can you provide an example of what you expect that'd look like?

File: src/Discovery/src/Eureka/DiscoveryClient.cs [hunk 1]
```diff
@@ -92,6 +92,9 @@ protected DiscoveryClient(ApplicationInfoManager appInfoManager, ILoggerFactory
             _startupLogger = logFactory?.CreateLogger("Startup." + this.GetType().FullName) ?? NullLogger.Instance;
         }
 
+        public event EventHandler<Applications> OnApplicationsChange;
+
```
Comment (TimHess) [inline src/Discovery/src/Eureka/DiscoveryClient.cs:96]:
```suggestion
```

**Comment by macsux** [comment]
Something like this
```csharp
public void Configure(IApplicationBuilder app, IWebHostEnvironment env, IDiscoveryClient client)
        {
            var eureka = (DiscoveryClient) client;
            eureka.OnApplicationsChange += (sender, applications) =>
            {
                Console.WriteLine("====");
                foreach (var application in applications.GetRegisteredApplications())
                {
                    Console.WriteLine($"App: {application.Name}");
                    foreach (var instance in application.Instances)
                    {
                        Console.WriteLine($"  {instance.Actiontype} {instance.HostName}:{instance.Port}");
                    }
                }
            };
```

**Comment by TimHess** [comment]
This implementation should be OK for now, but we'll probably want to revisit with [this issue](https://github.com/SteeltoeOSS/Steeltoe/issues/336) to align with Spring and things like [`ApplicationEvent`](https://github.com/spring-cloud/spring-cloud-commons/blob/main/spring-cloud-commons/src/main/java/org/springframework/cloud/client/discovery/event/HeartbeatEvent.java) 

Do you have time to add a unit test for this?

**Review by TimHess — APPROVED** [review]

**Pull request merged**